sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,500 @@
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+
+----
+
+`task_sequence` is a class template, parameterized by an `auto` reference to a
++Callable+ `f` and by a list of properties `propertiesT`.
+The +Callable+ `f` defines the asynchronous task to be
+associated with the `task_sequence`, and requiring an auto reference ensures
+that each `f` be statically resolvable at compile time.  Static resolvability by
+the compiler is desirable when compiling for spatial architectures as it can
+enable the generation of more efficient hardware. Furthermore, the partial
+specialization `typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...)`
+ensures that the return type `ReturnT` of `f` and the argument types
+`typename ... ArgsT` of `f` are resolvable and fixed for each definition of
+`task_sequence`. 
+
+The `task_sequence` interface consists of two member functions, `async` and
+`get`. The `async` function asynchronously invokes `f`, and stores the return
+value into a FIFO queue upon completion of `f`. By calling async on a
+`task_sequence` more than once, the user implies that the invocations of `f` can
+be run in parallel.
+
+The `async` function call is non-blocking in that it returns before the
+asynchronous `f` invocation completes executing, and potentially before `f` even
+begins executing (return from the `async` provides no implicit information on
+the execution status of `f`).
+
+The `get` member function retrieves the oldest result from this logical FIFO
+queue, and blocks (waits) until a result is available if none are available
+immediately upon the call to `get`. Both functions may only be invoked on the
+device on which a `task_sequence` object has been instantiated. Calling `async`
+or `get` on a different device results in undefined behavior.
+
+[frame="topbot",options="header"]
+|===
+|Functions |Description
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+task_sequence(const propertyListT &);
+
+task_sequence();
+----
+|
+Constructor for `task_sequence`. 
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+void async(ArgsT... args);
+----
+|
+Asynchronously calls +f+ with +args+. 
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+ReturnT get();
+----
+|
+Synchronously retrieves the result of an `async` call. 
+
+Results are retrieved in FIFO order of their `async` invocations. 
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+~task_sequence();
+----
+|
+Destructor for `task_sequence`. 
+
+Implicitly invokes `get` on all outstanding invocations launched through `async
+unless this `task_sequence` object was instantiated with the `balanced` property
+defined below.
+|===
+
+=== `task_sequence` Scoping
+
+To allow compilers targeting spatial compute architectures to efficiently reuse
+hardware, `task_sequence` objects must not have their lifetime extended beyond
+the scope in which they are declared, and it is undefined behavior if lifetime
+is otherwise extended. Both move and copy constructors for `task_sequence` are
```

Comment:
How would this be accomplished?  If it's not possible given the API then we don't even need to say that it's UB.

Reply:
I'm considering the case where you forget to call "get()" on a task_sequence that you've marked 'balanced' as a case of accessing an object that has gone out of scope, which is already illegal in C++, as opposed to a case of extending the lifetime of the object.

Reply:
Removed. Maybe it was written this way just to explain why move, copy ctors have been deleted.